p2p/chain.py

```diff
@@ -741,6 +633,144 @@ def _is_receipts_empty(header: BlockHeader) -> bool:
     return header.receipt_root == BLANK_ROOT_HASH
 
 
+async def _get_block_numbers_for_request(
+        headerdb: 'AsyncHeaderDB', block_number_or_hash: Union[int, bytes], max_headers: int,
+        skip: int, reverse: bool, token: CancelToken) -> List[BlockNumber]:
+    """
+    Generates the block numbers requested, subject to local availability.
+    """
+    block_number_or_hash = block_number_or_hash
+    if isinstance(block_number_or_hash, bytes):
+        header = await wait_with_token(
+            headerdb.coro_get_block_header_by_hash(cast(Hash32, block_number_or_hash)),
+            token=token,
+        )
+        block_number = header.block_number
+    elif isinstance(block_number_or_hash, int):
+        block_number = block_number_or_hash
+    else:
+        raise TypeError(
+            "Unexpected type for 'block_number_or_hash': %s",
+            type(block_number_or_hash),
+        )
+
+    limit = max(max_headers, eth.MAX_HEADERS_FETCH)
```

Comment:
What do you think about splitting this part of the body (the pure part) into a stand alone utility function for which we can write some simple tests.

Reply:
I'd prefer to save that for another PR, as I'm just shuffling this code around in this one